### ISSUE TITLE
"Query error: -1 Request was aborted" regression with 9.1.0-beta1; graphite datasource

### ISSUE THREAD

**Issue by ianw** [grafana/grafana#53275]
Hello

It seems the grafana:latest docker image updated to 9.1.0-beta1 recently (I'm not 100% sure this is intended, but it's how we saw this).

Some of our panels have started breaking with "Query error: -1 Request was aborted".  I'm not sure how useful it is, but in the web developer console we get an error:

```
runRequest.catchError {"type":"cancelled","cancelled":true,"data":null,"status":-1,"statusText":"Request was aborted","config":{"method":"POST","url":"https://graphite.opendev.org/render","data":"target=scale(divideSeries(stats.gauges.afs.afs02_dfw_openstack_org.part.vicepa.used%2C%20stats.gauges.afs.afs02_dfw_openstack_org.part.vicepa.total)%2C%20100)&from=1659552968&until=1659574570&format=json&maxDataPoints=100","headers":{"Content-Type":"application/x-www-form-urlencoded"},"requestId":"OpenStack.panelId.4","inspect":{"type":"graphite"},"retry":0,"hideFromInspector":false}} react_devtools_backend.js:4026:25
    overrideMethod react_devtools_backend.js:4026
    S runRequest.ts:151
    RxJS 29
    internalFetch backend_srv.ts:127
    H ResponseQueue.ts:32
    RxJS 6
    H ResponseQueue.ts:37
    _ FetchQueueWorker.ts:56
    RxJS 21
    B FetchQueue.ts:76
    B FetchQueue.ts:55
    RxJS 6
    B FetchQueue.ts:59
    fetch backend_srv.ts:112
    RxJS 69
    pipeToSubject PanelQueryRunner.ts:290
    run PanelQueryRunner.ts:271
    runAllPanelQueries PanelModel.ts:333
    se PanelChrome.tsx:321
    RxJS 3
    n EventBus.ts:37
    emit index.js:181
    publish EventBus.ts:27
    refresh PanelModel.ts:352
    panelInitialized DashboardModel.ts:375
    componentDidMount PanelChrome.tsx:180
    React 2
    unstable_runWithPriority scheduler.production.min.js:18
    React 4
    unstable_runWithPriority scheduler.production.min.js:18
    React 4
    Redux 6
    d actions.ts:37
    Redux 2
    ue DashboardPanel.tsx:68
    c LazyLoader.tsx:24
    observer LazyLoader.tsx:58
    (Async: IntersectionCallback)
    26346 LazyLoader.tsx:55
    Webpack 14
```

![Screenshot from 2022-08-04 10-58-21](https://user-images.githubusercontent.com/108448/182740480-7eff0607-da30-4fb4-bb09-8ca99f9f84f0.png)

I have not been able to establish exactly causes this, but it does seem to be related to graphs that are using ref-id's.  This graph above shows disk usage, so it has one target that gets the total space, and then the other divides the current used space using it's refId ```A```, e.g.

```json
"targets": [
    {
      "hide": true,
      "refId": "A",
      "target": "stats.gauges.afs.afs02_dfw_openstack_org.part.vicepa.total",
      "datasource": {
        "uid": "d9c17c45a8",
        "type": "graphite"
      }
    },
    {
      "hide": false,
      "refId": "B",
      "target": "scale(divideSeries(stats.gauges.afs.afs02_dfw_openstack_org.part.vicepa.used, #A), 100)",
      "datasource": {
        "uid": "d9c17c45a8",
        "type": "graphite"
      }
    }
  ],
```

I have put some of our CI that uses this on hold so it can be inspected at https://23.253.56.172/ (note need to accept self signed certificate as it is a test system).

The panel above is https://23.253.56.172/d/9871b26303/afs?orgId=1&viewPanel=4 which exhibits the problem

https://23.253.56.172/d/9871b26303/afs?orgId=1 is the overall graph that shows some of the panels working, and others showing this error.

Everything the same works with the 9.0.6 image, so it does seem to be something introduced between then and now.

**Comment by ianw** [comment]
I've bisected this down to 6c3efb0c88201ff4ace29c85e2dbb9a1f8b48f5b 

cc @bohandley

**Comment by zuchka** [comment]
Thanks for creating this issue, @ianw 

Here is a nice Influx2 dashboard on our public sandbox: https://play.grafana.org/d/2N0n0Er7k/influx-2-2-live-noaa-buoy-data?orgId=1

do you think you could try and replicate your issue on this instance? :pray:

**Comment by bohandley** [comment]
Hi @ianw, thank you for bringing this to my attention! This is definitely a regression for the ref-ids from my graphite annotation migration from angular to react. I have a fix for this and work to get it in v9.1.0-beta2

**Comment by melroy89** [comment]
Bug is present again in Grafana with InfluxDB using the AnnoListPanel.tsx.

I used to see this error just a handful of times Grafana v11.2.2:

![Image](https://github.com/user-attachments/assets/37d16839-1134-4ba3-8885-1ad6c07ad475)


However, this is getting absurd, and this annotation is called several times per second across all these graphs or whatever it is triggering, causing a lot of annotation request aborted and **slow graphs loading** (from influxdb) since Grafana 11.3 and 11.4.0.

When I try to manually resend the API calls (eg `https://myserver.com/api/annotations?tags=server&limit=50&type=annotation`), everything works fine nothing is wrong.

However in the latest Grafana version its getting ridiculously slow now and the amount of errors increased dramatically:

![Image](https://github.com/user-attachments/assets/0c3ee5b6-98d3-47be-846a-6b1dfc3ab388)


```
Uncaught (in promise) 
Object { type: "cancelled", cancelled: true, data: null, status: -1, statusText: "Request was aborted", config: {…} }
​
cancelled: true
​
config: Object { method: "GET", url: "api/annotations", requestId: "anno-list-panel-413", … }
​​
headers: Object { "X-Grafana-Org-Id": 1, "X-Grafana-Device-Id": "5a3c4ee35b92c7195d1f871ba19dd799" }
​​
hideFromInspector: true
​​
method: "GET"
​​
params: Object { tags: (1) […], limit: 50, type: "annotation" }
​​
requestId: "anno-list-panel-413"
​​
retry: 0
​​
url: "api/annotations"
​​
<prototype>: Object { … }
​
data: null
​
status: -1
​
statusText: "Request was aborted"
​
type: "cancelled"
​
<prototype>: Object { … }
​​
__defineGetter__: function __defineGetter__()
​​
__defineSetter__: function __defineSetter__()
​​
__lookupGetter__: function __lookupGetter__()
​​
__lookupSetter__: function __lookupSetter__()
​​
__proto__: 
​​
constructor: function Object()
​​
hasOwnProperty: function hasOwnProperty()
​​
isPrototypeOf: function isPrototypeOf()
​​
propertyIsEnumerable: function propertyIsEnumerable()
​​
toLocaleString: function toLocaleString()
​​
toString: function toString()
​​
valueOf: function valueOf()
​​
<get __proto__()>: function __proto__()
​​
<set __proto__()>: function __proto__()
AnnoListPanel.tsx:128:46
    doSearch AnnoListPanel.tsx:128
    AsyncFunctionThrow self-hosted:808
    (Async: async)
    componentDidMount AnnoListPanel.tsx:55
    React 7
    O scheduler.production.min.js:13
    J scheduler.production.min.js:14
    (Async: EventHandlerNonNull)
    7463 scheduler.production.min.js:14
    Webpack 27
```
